server/datastore/mysql/app_configs.go

```diff
@@ -67,6 +84,30 @@ func (ds *Datastore) SaveAppConfig(ctx context.Context, info *fleet.AppConfig) e
 	})
 }
 
+func (ds *Datastore) insertOrReplaceConfigAsset(ctx context.Context, asset fleet.MDMConfigAsset) error {
+	assets, err := ds.GetAllMDMConfigAssetsByName(ctx, []fleet.MDMAssetName{asset.Name})
```

Comment:
should we use the writer for all of these? pretty edge case but there's a potential race (maybe we're already using it in `SaveAppConfig`?)